### ISSUE TITLE
Add simple line breaking support

### ISSUE THREAD

**Issue by khaledhosny** [HOST-Oman/libraqm#50]
Just the basics (breaking at spaces and other simple break opportunities), just good enough for basic Arabic and Latin needs.

**Comment by khaledhosny** [comment]
https://github.com/grigorig/ucdn/pull/7 should be helpful here. cc @mzn723.

**Comment by brawer** [comment]
Hm, when I use this in [fontview](https://github.com/googlei18n/fontview), raqm enters an endless loop inside `_raqm_line_break()`. Specifically, in the loop `for (size_t i = 0; i < glyph_count; i++)` after the comment `/* finding possible breaks in text */`, the variable `i` stays at value `1` for every iteration. It gets of course increased on each loop iteration, but then it gets decreased again inside the loop. — To reproduce, check out fontview, uncomment the call to `raqm_set_line_width()` in `src/fontview/sample_text.cpp` and run `git checkout linebreaking` inside `src/third_party/raqm/libraqm`.

**Comment by khaledhosny** [comment]
I was debugging that a couple of weeks ago, and the current code in `_raqm_line_break()` is very fragile and I need to re-write it in a more robust way.

**Comment by brawer** [comment]
Thanks, much appreciated.

**Comment by brawer** [comment]
Curious, how’s the rewrite going?

**Comment by khaledhosny** [comment]
Not much unfortunately, didn’t have time to do Raqm hacking lately.

**Comment by brawer** [comment]
Curious, any updates? (Sorry for the nagging).

**Comment by khaledhosny** [comment]
No, unfortunately. I have been pretty busy with other stuff and right now I don’t know when I’ll have time again to look into this.

**Comment by ericoporto** [comment]
Hey, I am looking into this library because I wanted to use it along with Harfbuzz and FreeType for Persian, but I am a bit stuck at linebreaks, has there been any change in this issue?

**Comment by khaledhosny** [comment]
No one is actively working on this unfortunately, and I cam to the conclusion that this is best handled by an more actively maintained library, probably libunibreak than trying to replicate that work here, though I suspect some Raqm users might not appreciate the extra dependency.

**Comment by therahedwig** [comment]
> No one is actively working on this unfortunately, and I cam to the conclusion that this is best handled by an more actively maintained library, probably libunibreak than trying to replicate that work here, though I suspect some Raqm users might not appreciate the extra dependency.

Hi, I was wondering, might it be an idea to have sort of wrapper functions around the linebreaking functionality, that library-consumers can then reimplement, much like [harfbuzz](https://harfbuzz.github.io/customizing-unicode-functions.html) has for it's unicode functions? This way this library can have some super-basic implementation for people who don't need much, and then people who need up-to-date linebreaking algos can use libunibreak or ICU directly.
